baseplate/lib/secrets.py

```diff
@@ -116,24 +118,68 @@ def _decode_secret(path: str, encoding: str, value: str) -> bytes:
     raise CorruptSecretError(path, f"unknown encoding: {encoding!r}")
 
 
+class SecretParser:
+    @staticmethod
+    def get_secret(data: Dict[str, Any], secret_path: str = "") -> Dict[str, str]:
+        raise NotImplementedError
+
+
+class VaultFileParser(SecretParser):
+    @staticmethod
+    def get_secret(data: Dict[str, Any], secret_path: str = "") -> Dict[str, str]:
+        try:
+            return data["secrets"][secret_path]
+        except KeyError:
+            raise SecretNotFoundError(secret_path)
+
+
+class VaultCSIParser(SecretParser):
+    @staticmethod
+    def get_secret(data: Dict[str, Any], secret_path: str = "") -> Dict[str, str]:
+        return data["data"]
```

Comment:
These don't need to be classes. It can just be a function:

```suggestion
SecretParser = Callable[[Dict[str, Any], str], Dict[str, str]]


def parse_secrets_fetcher(data: Dict[str, Any], secret_path: str = "") -> Dict[str, str]:
    try:
        return data["secrets"][secret_path]
    except KeyError:
        raise SecretNotFoundError(secret_path)


def parse_vault_csi(data: Dict[str, Any], secret_path: str = "") -> Dict[str, str]:
    return data["data"]
```